src/Processors/Transforms/JoiningTransform.cpp

```diff
@@ -400,6 +420,38 @@ void DelayedJoinedBlocksWorkerTransform::work()
     output_chunk.setColumns(block.getColumns(), rows);
 }
 
+void DelayedJoinedBlocksWorkerTransform::resetTask()
+{
+    task.reset();
+    left_delayed_stream_finished = false;
+    setup_non_joined_stream = false;
+    non_joined_delayed_stream = nullptr;
+}
+
+Block DelayedJoinedBlocksWorkerTransform::nextNonJoinedBlock()
+{
+    if (!setup_non_joined_stream)
```

Comment:
AFAIU, `setup_non_joined_stream` flag is a try to reduce possible contention, right? Logically, it seems not necssary